query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/simple/find_first_or_throw.rs

```diff
@@ -0,0 +1,69 @@
+use query_engine_tests::*;
+
+#[test_suite(schema(schemas::generic))]
+mod find_first_or_throw_query {
+    use query_engine_tests::assert_query;
+
+    #[connector_test]
+    async fn find_first_or_throw_matching(runner: Runner) -> TestResult<()> {
```

Comment:
Test case to ensure that `findFirstOrThrow` behaves like `findFirst` when a record is found